File: src/main/java/me/lokka30/levelledmobs/result/PlayerHomeCheckResult.java
package me.lokka30.levelledmobs.result;

import org.bukkit.Location;
import org.jetbrains.annotations.Nullable;

/**
 * Used to hold info that is used for various
 * custom placeholders
 *
 * @author stumper66
 * @since 3.6.0
 */
public class PlayerHomeCheckResult {

    public @Nullable Location location;
    public @Nullable
    final String resultMessage;
    public @Nullable String homeNameUsed;

    public PlayerHomeCheckResult(final @Nullable String resultMessage) {
        this.resultMessage = resultMessage;
    }

    public PlayerHomeCheckResult(final @Nullable String resultMessage,
        final @Nullable Location location) {
        this.resultMessage = resultMessage;
        this.location = location;
    }

    public PlayerHomeCheckResult(final @Nullable String resultMessage,
        final @Nullable Location location, final @Nullable String homeNameUsed) {
        this.resultMessage = resultMessage;
        this.location = location;
        this.homeNameUsed = homeNameUsed;
    }
}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/result/PlayerHomeCheckResult.kt
package io.github.arcaneplugins.levelledmobs.result

import org.bukkit.Location

/**
 * Used to hold info that is used for various
 * custom placeholders
 *
 * @author stumper66
 * @since 3.6.0
 */
class PlayerHomeCheckResult {
    var location: Location? = null
        private  set
    var resultMessage: String? = null
        private set
    var homeNameUsed: String? = null
        private set

    constructor(resultMessage: String?){
        this.resultMessage = resultMessage
    }

    constructor(resultMessage: String?, location: Location?){
        this.resultMessage = resultMessage
        this.location = location
    }

    constructor(resultMessage: String?, location: Location?, homeNameUsed: String?){
        this.resultMessage = resultMessage
        this.location = location
        this.homeNameUsed = homeNameUsed
    }
}